hypervisor/dm/vpci/vroot_port.c

```diff
@@ -140,14 +141,19 @@ int32_t create_vrp(struct acrn_vm *vm, struct acrn_vdev *dev)
 			dev_config->vdev_ops = &vrp_ops;
 
 			vdev = vpci_init_vdev(&vm->vpci, dev_config, NULL);
+			if (vdev == NULL) {
+				pr_err("%s: failed to create virtual root port\n", __func__);
+				ret = -EFAULT;
+				break;
+			}
```

Comment:
There are two other calls to vpci_init_vdev() without checks on the returned pointer, one in create_vmcs9900_vdev() and the other in create_ivshmem_vdev(). Both can be called via hypercalls, and thus require the errors to be reported.